flexget/api.py

```diff
@@ -0,0 +1,635 @@
+
+
+@api.doc(params={'exec_id': 'Execution ID of the Task'})
+@api.doc(description='Execution ID are held in memory, they will be lost upon daemon restart')
+@execution_api.route('/<exec_id>/')
+class ExecutionTaskAPI(APIResource):
+
+    @api.response(404, 'task execution not found')
+    @api.response(200, 'list of tasks queued for execution', task_execution_api_schema)
+    def get(self, exec_id, session=None):
+        """ Status of existing task execution """
+        task_info = manager.task_queue.tasks_info.get(exec_id)
+
+        if not task_info:
+            return {'error': '%s not found' % exec_id}, 404
+
+        return _task_info_dict(task_info)
+
+
+@api.doc(params={'exec_id': 'Execution ID of the Task'})
+@api.doc(description='Execution ID are held in memory, they will be lost upon daemon restart')
+@execution_api.route('/<exec_id>/log/')
+class ExecutionTaskLogAPI(APIResource):
+    @api.response(200, 'Streams as line delimited JSON')
+    @api.response(404, 'task log not found')
+    def get(self, exec_id, session=None):
+        """ Log stream of executed task
+        Streams as line delimited JSON
+        """
+        task_info = manager.task_queue.tasks_info.get(exec_id)
+
+        if not task_info:
+            return {'error': '%s not found' % exec_id}, 404
+
+        def follow():
+            f = open(os.path.join(manager.config_base, 'log-%s.json' % manager.config_name), 'r')
+            while True:
+                if not task_info.started:
+                    continue
+
+                # First check if it has finished, if there is no new lines then we can return
+                finished = task_info.finished is not None
+                line = f.readline()
+                if not line:
+                    if finished:
+                        return
+                    sleep(0.1)
+                    continue
+
+                record = json.loads(line)
+                if record['task_id'] != exec_id:
+                    continue
+                yield line
+
+        return Response(follow(), mimetype='text/event-stream')
+
+
+# Tasks API
+tasks_api = api.namespace('tasks', description='Manage Tasks')
+
+task_api_schema = {
+    'type': 'object',
+    'properties': {
+        'name': {'type': 'string'},
+        'config': {'$ref': '/schema/plugins'}
+    }
+}
+
+tasks_api_schema = {
+    "type": "object",
+    "properties": {
+        "tasks": {
+            "type": "array",
+            "items": task_api_schema
+        }
+    }
+}
+
+tasks_api_schema = api.schema('tasks', tasks_api_schema)
+task_api_schema = api.schema('task', task_api_schema)
+
+
+def _load_config():
+    with codecs.open(manager.config_path, 'rb', 'utf-8') as f:
+        try:
+            raw_config = f.read()
+        except UnicodeDecodeError:
+            raise ValueError('Config file is not UTF-8 encoded')
+    return yaml.safe_load(raw_config) or {}
```

Comment:
Yeah, and the config is already in memory at `manager.config` that's where everything gets its config from ultimately.